### PR TITLE
mitigate unix socket path problem

### PR DESCRIPTION
**Pull request by huangminghuang**
This PR mitigates unix socket path too long problem for http plugin

Related to https://github.com/AntelopeIO/leap/issues/840

### CODE REVIEW

File: plugins/http_plugin/http_plugin.cpp
Comment (linh2931) [inline plugins/http_plugin/http_plugin.cpp:280]: Probably just wait until your std::filesystem is merged before doing this, such that you don't need to change boost::filesystem to std::filesystem one more time.

You should still check if the relative path is longer than  unix_socket_limit, and log you are trying a relative path.

File: plugins/http_plugin/http_plugin.cpp
Comment (huangminghuang) [reply plugins/http_plugin/http_plugin.cpp:281]: The motivation of the PR is to avoid testing failing when the test log directories are too long to begin with. Just let user to re-try with a relative path doesn't help the problem.

File: plugins/http_plugin/http_plugin.cpp
Comment (huangminghuang) [reply plugins/http_plugin/http_plugin.cpp:280]: If the relative is longer than the limit, we would get a system exception with a 'path is too long` description already. Extra checking doesn't do anything better.

File: plugins/http_plugin/http_plugin.cpp
Comment (heifner) [reply plugins/http_plugin/http_plugin.cpp:281]: I like Matt's idea. Then it would always work the same way and could never fail do to length issue.

**Review by heifner — APPROVED** [review]

**Review by linh2931 — APPROVED** [review]

**Pull request merged**
